rust/migrate-wicked/src/vlan.rs

```diff
@@ -0,0 +1,97 @@
+use agama_dbus_server::network::model::{self};
+use serde::{Deserialize, Serialize};
+use serde_with::{skip_serializing_none, DeserializeFromStr, SerializeDisplay};
+use strum_macros::{Display, EnumString};
+
+#[derive(
+    Debug, Clone, PartialEq, SerializeDisplay, DeserializeFromStr, EnumString, Display, Default,
+)]
+#[strum(serialize_all = "kebab_case")]
+pub enum VlanProtocol {
+    #[default]
+    #[strum(serialize = "ieee802-1Q")]
+    Ieee802_1Q,
+    Ieee802Ad,
+}
+
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+#[skip_serializing_none]
+pub struct Vlan {
+    pub device: String,
+    pub address: Option<String>,
+    #[serde(default)]
+    pub protocol: VlanProtocol,
+    pub tag: u16,
+}
+
+impl From<&Vlan> for model::ConnectionConfig {
+    fn from(vlan: &Vlan) -> model::ConnectionConfig {
+        model::ConnectionConfig::Vlan(model::VlanConfig {
+            parent: vlan.device.clone(),
+            id: (vlan.tag as u32),
+            protocol: (&vlan.protocol).into(),
+        })
+    }
+}
+
+impl From<&VlanProtocol> for model::VlanProtocol {
+    fn from(v: &VlanProtocol) -> model::VlanProtocol {
+        match v {
+            VlanProtocol::Ieee802_1Q => model::VlanProtocol::IEEE802_1Q,
+            VlanProtocol::Ieee802Ad => model::VlanProtocol::IEEE802_1ad,
```

Comment:
I always think this is confusing so again I would differentiate at least the wicked part with `WickedVlanProtocol` and maybe `AgamaVlanProtocol`, but just personal taste. Keep it if you don't think it's an issue.

Reply:
I changed to `WickedVlanProtocol`. But I keept the `model::VlanProtocol` as I think this `model::` prefix is already naming enough.